static/purchase.js

```diff
@@ -0,0 +1,63 @@
+// ----- Setup ----- //
+
+const DRIVER_TOTAL = 1000;
+const TEAM_TOTAL = 1000;
+
+
+// ----- Functions ----- //
+
+// Retrieves the drivers and teams from the page.
+function getElems () {
+
+    return {
+	drivers: document.querySelectorAll('.driver-table input[type=checkbox]:checked'),
+	teams: document.querySelectorAll('.team-table input[type=checkbox]:checked')
+    };
+    
+}
+
+// Sums a list of objects with a price field.
+function sumPrices (total, {price}) {
+    return total + parseInt(price);
+}
+
+// Calculates whether the spend is within budget.
+function withinBudget (drivers, teams) {
+
+    let driverTotal = drivers.reduce(sumPrices, 0);
```

Comment:
Yeah true, but I think I made it a separate function because it's used a couple times, and because I figured the name `sumPrices` made it a little clearer what's going on. The second point I guess is very much a judgement call.

Also, that's not quite equivalent, you'd have to throw in a couple of `parseInts` =(.